pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -279,24 +279,51 @@ stdenv.mkDerivation {
       export hardening_unsupported_flags="${builtins.concatStringsSep " " (cc.hardeningUnsupportedFlags or [])}"
     ''
 
-    + optionalString hostPlatform.isCygwin ''
-      hardening_unsupported_flags+=" pic"
-    ''
+    # Machine flags. These are necessary to support
 
-    + optionalString targetPlatform.isMinGW ''
-      hardening_unsupported_flags+=" stackprotector"
+    # TODO: We should make a way to support miscellaneous machine
+    # flags and other gcc flags as well.
+
+    # Always add -march based on cpu in triple. Sometimes there is a
+    # discrepency (x86_64 vs. x86-64), so we provide an "arch" arg in
+    # that case.
+    + optionalString (targetPlatform ? platform.gcc.arch || targetPlatform.parsed.cpu ? arch) ''
+      echo "-march=${targetPlatform.platform.gcc.arch or targetPlatform.parsed.cpu.arch}" >> $out/nix-support/cc-cflags-before
```

Comment:
In some cases ```-march``` is not enough to specify allowed opcodes.
For example [AVX-512 has optional extentions ](https://en.wikipedia.org/wiki/AVX-512#Instruction_set) and in addicion to -march=skylake-avx512 there should be set of  -mavx5124fmaps -mavx5124vnniw -mavx512bitalg -mavx512bw -mavx512cd -mavx512dq -mavx512er -mavx512f -mavx512ifma -mavx512pf -mavx512vbmi -mavx512vbmi2 -mavx512vl -mavx512vnni -mavx512vpopcntdq to enable particular extentions

Reply:
So both ```platform.gcc.arch``` and ```platform.gcc.tune``` need to allow more than one command line flag.
It could be hacked around with ```platform.gcc.arch = "skylake-avx512\n-mavx5124vnniw";``` but it is weird.